src/theming/mixins/listItem.ts

```diff
@@ -0,0 +1,84 @@
+export interface ListItemMixinStates {
+  /** The background of the element. */
+  background: string;
+  /** The color text of the element. */
+  color: string;
+  /** The transitionDuration of the ripple effect. */
+  transitionDuration?: number;
+  /** The animationName of the ripple effect. */
+  animationName?: string;
+  /** The background if element is hover. */
+  hover: string;
+  /** The ripple effect background if element has focus. */
+  focus: string;
+  /** The ripple effect background if element is active. */
+  active: string;
+  /** The selected element background. */
+  backgroundSelected?: string;
+  /** The selected element background if hover. */
+  hoverSelected?: string;
+  /** The ripple effect background if selected element has focus. */
+  focusSelected?: string;
+  /** The ripple effect background if selected element is active. */
+  activeSelected?: string;
+}
+
+export const listItemMixin = (states: ListItemMixinStates) => {
+  const {
+    background,
+    color,
+    transitionDuration,
+    animationName,
+    hover,
+    focus,
+    active,
+    backgroundSelected,
+    hoverSelected,
+    focusSelected,
+    activeSelected
+  } = states;
+
+  return {
+    '&, &:hover, &:active': {
+      backgroundColor: background,
+      color: color
+    },
+    '& .MuiTouchRipple-root': {
+      transitionDuration: `${transitionDuration}ms`,
+      color: active
+    },
+    '& .MuiTouchRipple-rippleVisible': {
+      animationName: `${animationName} !important`,
+      opacity: '1 !important'
+    },
+    '@media (hover: hover)': {
+      '&:not(.Mui-focusVisible):hover': {
```

Comment:
Наверно тут стоит немного переделать логику и менять фон для .MuiTouchRipple-root, как при selected.
```suggestion
      '&:not(.Mui-focusVisible):hover .MuiTouchRipple-root': {
```